### PR TITLE
Add support for string size mechanism of passed-in

### LINKED ISSUE #900: GetAStringWithSpecifiedMaximumSize does not exist in nifake's 'functions.py' but is called out in functions_addon.py
GetAStringWithSpecifiedMaximumSize does not exist in functions.py

https://github.com/ni/nimi-python/blob/0670ca9d755be3a4ab96d9f431914cd92829b2e1/src/nifake/metadata/functions_addon.py#L48

- The generator should have caught this.
- We are not testing this mechanism.

### PR DESCRIPTION
**Pull request by texasaggie97**
- [X] This contribution adheres to [CONTRIBUTING.md](https://github.com/ni/nimi-python/blob/master/CONTRIBUTING.md).
- [X] ~~I've updated [CHANGELOG.md](https://github.com/ni/nimi-python/blob/master/CHANGELOG.md) if applicable.~~
- [X] I've added tests applicable for this pull request
### What does this Pull Request accomplish?
* Add support for string size mechanism of `passed-in`
* Add test in nifake to use new mechanism
* For some reason, after re-installing Windows, flake8 (of any version) would not look in `tox.ini` without telling it that that was the config file. I don't believe this will hurt and may keep this from happening to others.
### List issues fixed by this Pull Request below, if any.
* Fix #900 

### What testing has been done?
* Unit

### CODE REVIEW

File: generated/nifake/session.py [hunk 1]
```diff
@@ -955,6 +955,27 @@ def get_a_string_using_python_code(self, a_number):
         errors.handle_error(self, error_code, ignore_warnings=False, is_error_handling=False)
         return a_string_ctype.value.decode(self._encoding)
 
+    @ivi_synchronized
+    def get_a_string_with_specified_maximum_size(self, buffer_size):
+        r'''get_a_string_with_specified_maximum_size
+
+        Returns multiple types.
```
Comment (marcoskirsch) [inline generated/nifake/session.py]: Is this right? Seems like a bug in the generator (if so, should be addressed in its own PR)

File: src/nifake/metadata/functions.py [hunk 2]
```diff
@@ -923,6 +923,43 @@
...
+                    'description': 'Number of bytes allocated for aString',
+                },
+            },
+            {
+                'direction': 'out',
+                'enum': None,
+                'name': 'aString',
+                'type': 'ViChar[]',
+                'documentation': {
+                    'description': 'An string with passed in size.',
```
Comment (marcoskirsch) [inline src/nifake/metadata/functions.py]: An -> A

File: src/nifake/metadata/functions.py [hunk 3]
```diff
@@ -923,6 +923,43 @@
...
+                },
+            },
+        ],
+        'documentation': {
+            'description': 'Returns multiple types.',
```
Comment (marcoskirsch) [inline src/nifake/metadata/functions.py]: Please fix. It through this was created by the generator ;)

File: generated/nifake/session.py [hunk 4]
```diff
@@ -955,6 +955,27 @@ def get_a_string_using_python_code(self, a_number):
         errors.handle_error(self, error_code, ignore_warnings=False, is_error_handling=False)
         return a_string_ctype.value.decode(self._encoding)
 
+    @ivi_synchronized
+    def get_a_string_with_specified_maximum_size(self, buffer_size):
```
Comment (epage) [inline generated/nifake/session.py]: Why is a python user expected to pass in a buffer size?  That feels like a C API bleeding over into Python.  Is there really no way we can work around this?

**Comment by marcoskirsch** [comment]
Here's an additional problem: If we decide we absolutely need this in out public API:

```python
def get_a_string_with_specified_maximum_size(self, buffer_size):
 ```

We may or may not have our string-encoding story 100% correct yet and we get away with it becuase the APIs tend to return ASCII only. But that assumption will not hold forever. So the number passed in by the client (buffer_size) may not match the number of characters from the client's point of view when the API returns non-ASCII characters.

**Comment by texasaggie97** [comment]
This is becoming too much for a mechanism we haven't needed so far.

I propose abandoning this PR and not adding this mechanism. Instead see #950 for how to close #900

**Comment by marcoskirsch** [comment]
> This is becoming too much for a mechanism we haven't needed so far.

Furthermore, I don't think we'd ever want to expose it in this manner; at least not in the public API. I can see ourselves maybe using it in a private method?

I'm ok abandoning altogether.
